### PR TITLE
Suggestions

### PR DESCRIPTION
**Pull request by DaveCTurner**

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java [hunk 1]
```diff
@@ -76,7 +76,7 @@
 import static org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin;
 import static org.elasticsearch.xpack.ml.utils.InferenceProcessorInfoExtractor.pipelineIdsByResource;
 
-public class TransportGetTrainedModelsStatsAction extends HandledTransportAction<
+public class TransportGetTrainedModelsStatsAction extends TransportAction<
```
Comment (DaveCTurner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java:79]: No need to be a `HandledTransportAction`, this action is only ever invoked on the local node AFAICT.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java [hunk 2]
```diff
@@ -114,6 +108,15 @@ protected void doExecute(
         Task task,
         GetTrainedModelsStatsAction.Request request,
         ActionListener<GetTrainedModelsStatsAction.Response> listener
+    ) {
+        // workaround for https://github.com/elastic/elasticsearch/issues/97916 - TODO remove this when we can
+        executor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)));
```
Comment (DaveCTurner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java:113]: fork the whole thing onto `executor` first

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java [hunk 3]
```diff
@@ -122,90 +125,111 @@ protected void doExecute(
...
+                    request.isAllowNoResources(),
+                    request.getPageParams(),
+                    Collections.emptySet(),
+                    modelAliasMetadata,
+                    parentTaskId,
+                    matchedDeploymentIds,
+                    l
+                );
+            })
+            .andThenAccept(tuple -> responseBuilder.setExpandedModelIdsWithAliases(tuple.v2()).setTotalModelCount(tuple.v1()))
```
Comment (DaveCTurner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java:151]: suggest using `andThenAccept` to split out the handling of the previous step's result from the initiation of the next steps.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java [hunk 4]
```diff
@@ -122,90 +125,111 @@ protected void doExecute(
...
-            );
-            getDeploymentStats(client, request.getResourceId(), parentTaskId, assignmentMetadata, l);
-        }).<Map<String, TrainedModelSizeStats>>andThen(executor, null, (l, deploymentStats) -> {
-            // deployment stats for each matching deployment
-            // not necessarily for all models
-            responseBuilder.setDeploymentStatsByDeploymentId(
-                deploymentStats.getStats()
-                    .results()
+        SubscribableListener
+
```
Comment (DaveCTurner) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java:129]: More blank lines makes for better formatting throughout.

**Comment by droberts195** [comment]
Thanks @DaveCTurner!

**Pull request merged**